denonavr/api.py

```diff
@@ -316,134 +318,210 @@ def is_default_async_client(self) -> bool:
     timeout: float = attr.ib(converter=float, default=2.0)
+    _connection_enabled: bool = attr.ib(default=False)
     _healthy: Optional[bool] = attr.ib(
         converter=attr.converters.optional(bool),
         default=None)
+    _last_message_time: float = attr.ib(default=-1.0)
     _connect_lock: asyncio.Lock = attr.ib(default=attr.Factory(asyncio.Lock))
-    _monitor_task: asyncio.Task = attr.ib(default=None)
-    _reader: asyncio.StreamReader = attr.ib(default=None)
-    _writer: asyncio.StreamWriter = attr.ib(default=None)
+    _reconnect_task: asyncio.Task = attr.ib(default=None)
+    _monitor_handle: asyncio.TimerHandle = attr.ib(default=None)
+    _protocol: DenonAVRTelnetProtocol = attr.ib(default=None)
     _callbacks: Dict[str, Callable] = attr.ib(
         validator=attr.validators.instance_of(dict),
         default=attr.Factory(dict), init=False)
 
     async def async_connect(self) -> None:
         """Connect to the receiver asynchronously."""
+        _LOGGER.debug("%s: telnet connecting", self.host)
         async with self._connect_lock:
             if self.connected is True:
                 return
-            try:
-                self._reader, self._writer = await asyncio.wait_for(
-                    asyncio.open_connection(self.host, 23),
-                    timeout=self.timeout)
-            except asyncio.TimeoutError as err:
-                _LOGGER.debug("Timeout exception on telnet connect")
-                raise AvrTimoutError(
-                    "TimeoutException: {}".format(err),
-                    "telnet connect") from err
-            except ConnectionRefusedError as err:
-                _LOGGER.debug(
-                    "Connection refused on telnet connect", exc_info=True)
-                raise AvrNetworkError(
-                    "ConnectionRefusedError: {}".format(err),
-                    "telnet connect") from err
-            except (OSError, IOError) as err:
-                _LOGGER.debug(
-                    "Connection failed on telnet reconnect", exc_info=True)
-                raise AvrNetworkError(
-                    "OSError: {}".format(err), "telnet connect") from err
-            self._healthy = True
-            self._monitor_task = asyncio.create_task(self._async_monitor())
+            await self._async_establish_connection()
+
+    async def _async_establish_connection(self) -> None:
+        """Establish a connection to the receiver."""
+        loop = asyncio.get_event_loop()
+        _LOGGER.debug("%s: establishing telnet connection", self.host)
+        try:
+            async with async_timeout.timeout(self.timeout):
+                transport_protocol = await loop.create_connection(
+                    lambda: DenonAVRTelnetProtocol(
+                        on_connection_lost=self._disconnected,
+                        on_message=self._process_event
+                    ),
+                    self.host,
+                    23
+                )
```

Comment:
```suggestion
            transport_protocol = await asyncio.wait_for(loop.create_connection(
                lambda: DenonAVRTelnetProtocol(
                    on_connection_lost=self._disconnected,
                    on_message=self._process_event
                ),
                self.host,
                23
            ), timeout=self.timeout)
```
As mentioned before, please remove `async_timeout` dependency. It is easy to replace this call by standard python expressions.

Reply:
Not relevant anymore